src/pick.ts

```diff
@@ -1,23 +1,24 @@
-function get (object: unknown, path: Array<string>): unknown {
-  return path.reduce((obj: Record<string, unknown>, p: string) => {
-    return obj?.[p] as Record<string, unknown>
-  }, object as Record<string, unknown>)
+import type { StateTree } from 'pinia'
+
+function get (object: StateTree, path: string[]) {
+  return path.reduce((obj: StateTree, p: string) => {
+    return obj?.[p] as StateTree
+  }, object)
 }
 
-function set (object: unknown, path: Array<string>, val: unknown): unknown {
+function set (object: StateTree, path: string[], val: unknown) {
```

Comment:
keep types as they are rn, will eventually improve them in a separate dev

Reply:
Do you want me to make a separate PR?